internal/rollup/terser-wrapped.js

```diff
@@ -74,51 +111,99 @@ function runterser(inputFile, outputFile, sourceMapFile) {
+/**
+ *
+ * @param {{inputFile: string, outputFile: string, sourceMapFile: string}[]} files
+ * @param {boolean} debug
+ */
+async function runTerserSerial(files, debug) {
+  const Terser = require(TERSER_MODULE_LOCATION);
+  for (const f of files) {
+    // run one set of terser at a time
+    await runTerser(f.inputFile, f.outputFile, f.sourceMapFile, Terser, debug);
   }
+}
 
-  fs.writeFileSync(config, JSON.stringify(terserConfig));
-
-  const args = [
-    require.resolve('build_bazel_rules_nodejs_rollup_deps/node_modules/terser/bin/uglifyjs'),
-    inputFile, '--output', outputFile, '--config-file', config
-  ];
+/**
+ *
+ * @param {{inputFile: string, outputFile: string, sourceMapFile: string}[]} files
+ * @param {boolean} debug
+ */
+async function runTerserParallel(files, debug) {
+  // TODO: make this configurable
+  const MAX_CONCURRENT_THREADS = os.cpus().length - 1;
+  const limit = pLimit(MAX_CONCURRENT_THREADS);
+
+  const concurrencyLimitedWorkers = files.map(
+      f => limit(() => runTerserThread(f.inputFile, f.outputFile, f.sourceMapFile, debug)));
+  await Promise.all(concurrencyLimitedWorkers);
+}
 
-  for (arg in argv) {
-    const prefix = arg.length == 1 ? '-' : '--';
-    const value = argv[arg];
-    args.push(prefix + arg);
-    if (value && value !== true) {
-      args.push(value);
-    }
-  }
+/**
+ *
+ * @param {string} inputFile
+ * @param {string} outputFile
+ * @param {string} sourceMapFile
+ * @param {boolean} debug
+ * @returns {Promise<void>}
+ */
+function runTerserThread(inputFile, outputFile, sourceMapFile, debug) {
+  return new Promise((resolve, reject) => {
+    const task = {
+      inputFile,
+      outputFile,
+      sourceMapFile,
+      terser: require.resolve(TERSER_MODULE_LOCATION),
+      debug
+    };
+    const worker = new worker_threads.Worker(__filename, {workerData: task});
+
+    worker.on('exit', () => resolve());
+    worker.on('error', err => reject(err));
+  })
+}
 
-  if (DEBUG) console.error(`Running node ${args.join(' ')}`);
+/**
+ *
+ * @param {string} inputFile
+ * @param {string} outputFile
+ * @param {string} sourceMapFile
+ * @param {Function} Terser
+ * @param {boolean} debug
+ * @returns {Promise<void>}
+ */
+async function runTerser(inputFile, outputFile, sourceMapFile, Terser, debug) {
+  if (DEBUG) console.error(`Minifying ${inputFile} -> ${outputFile} (sourceMap ${sourceMapFile})`);
 
-  const isWindows = /^win/i.test(process.platform);
-  child_process.execFileSync(
-      isWindows ? 'node.exe' : 'node', args,
-      {stdio: [process.stdin, process.stdout, process.stderr]});
+  const terserOptions = buildTerserConfig(sourceMapFile, debug)
+  const inputCode = (await fs.promises.readFile(inputFile)).toString();
+  const result = Terser.minify(inputCode, terserOptions);
+
+  if (result.error) {
+    // TODO: figure out error reporting here
+    // scroll up a little from this: https://github.com/terser-js/terser#minify-options
+    throw result.error;
```

Comment:
Im not certain of the format of this error object, we need to do some trail and error I think to figure out what kind of errors we get from terser.
Im not familiar enough with it to know how to cause an error :P